### ISSUE TITLE
Module polyfill override project polyfill

### ISSUE THREAD

**Issue by ZxBing0066** [zloirock/core-js#880]
**In the following case, Promise.finally will be wrongly overwritten**:

In project A I imported module B, the module bundled core-js with `buildIn: usage`, then it bundle polyfill with this in builded code:
```js
require("core-js/modules/es.promise");
```

In project A, I want to bundle all polyfill into one file, so I use `buildIn: entry` to bundle a single `polyfill.js`, then in my project bundle, bundle html will be this:
```html
<script src='polyfill.js'></script>
<script src='A.js'></script>
```

The polyfill.js load the right Promise, but in Edge 18, because of empty `PromiseRejectionEvent`, Promise will force override by `PromiseConstructor`:
```js
// node_modules/core-js/modules/es.promise.js:53
var FORCED = isForced(PROMISE, function () {
  var GLOBAL_CORE_JS_PROMISE = inspectSource(PromiseConstructor) !== String(PromiseConstructor);
  if (!GLOBAL_CORE_JS_PROMISE) {
    if (V8_VERSION === 66) return true;
    // PromiseRejectionEvent is undefined, FORCED will be true
    if (!IS_NODE && typeof PromiseRejectionEvent != 'function') return true;
  }
  // ...
});
```
It's OK now, but when run into A.js, B module bundled `core-js/modules/es.promise`, without `es.promise.finally`, then it's broken, the var `FORCED` always to be true, the `Promise` will be override with the next one without `finally`.

So **is there some good idea to avoid duplicated `Promise` polyfill load**?
Such as add a tag to Promise polyfill, then in next polyfill running, check the tag to avoid duplicated overridden?

**Comment by zloirock** [comment]
Yes, this feature detection test case is problematic. I have no ideas about how to detect unhandled rejection tracking **synchronously** with another way than check `PromiseRejectionEvent`. Do you have better ideas?

However, at first, your issue related to good practices of usage - you should load only one copy of the global `core-js` version.

Sure, good tooling could help to prevent problems like this. I have some ideas which could help to prevent problems like this in the next generation of tooling, but if it will be created - it will be created not soon.

**Comment by ZxBing0066** [comment]
> Yes, this feature detection test case is problematic. I have no ideas about how to detect unhandled rejection tracking support **synchronously** with another way than check `PromiseRejectionEvent`. Do you have better ideas?
> 
> However, at first, your issue related to good practices of usage - you should load only one copy of the global `core-js` version.
> 
> Sure, good tooling could help to prevent problems like this. I have some ideas which could help to prevent problems like this in the next generation of tooling, but if it will be created - it will be created not soon.

Yes, I know global single code-js is a good idea, but in some case, such as core-js loaded in some library or in some plugin, it's not uncontrolled by me.

Maybe we can add a custom property to the polyfilled `PromiseConstructor` such as `__FROM_CORE_JS__`, then in `if(FORCED)` check `__FROM_CORE_JS__` and avoid to override the current.

// core-js/modules/es.promise.js:225
```js
// check the current NativePromise should override
if (FORCED && !PromiseConstructor.__FROM_CORE_JS__) {
  PromiseConstructor = function Promise(executor) {
    // ...
  };
  // add the tag property
  PromiseConstructor.__FROM_CORE_JS__ = true;
}
```

**Comment by zloirock** [comment]
> Maybe we can add a custom property to the polyfilled `PromiseConstructor` such as `__FROM_CORE_JS__`, then in `if(FORCED)` check `__FROM_CORE_JS__` and avoid to override the current.

It's already here - `GLOBAL_CORE_JS_PROMISE`. Most likely one of your `core-js` versions is obsolete.

**Comment by ZxBing0066** [comment]
> > Maybe we can add a custom property to the polyfilled `PromiseConstructor` such as `__FROM_CORE_JS__`, then in `if(FORCED)` check `__FROM_CORE_JS__` and avoid to override the current.
> 
> It's already here - `GLOBAL_CORE_JS_PROMISE`. Most likely one of your `core-js` versions is obsolete.

No. My core-js version is `3.6.5`.

The `GLOBAL_CORE_JS_PROMISE` has no effect in this case, both of inspectSource(PromiseConstructor) and String(PromiseConstructor) return the same code.

```js
function Promise(executor) {
    anInstance(this, PromiseConstructor, PROMISE);
    aFunction(executor);
    Internal.call(this);
    var state = getInternalState(this);
    try {
      executor(bind(internalResolve, this, state), bind(internalReject, this, state));
    } catch (error) {
      internalReject(this, state, error);
    }
  };
```

**Comment by ZxBing0066** [comment]
It's fixed in 3.7.0, thank you.

😄
